app/web_framework/cluster_master_application.py

```diff
@@ -41,7 +42,8 @@ def __init__(self, cluster_master):
                     ]),
                     RouteNode(r'queue', _QueueHandler),
                     RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
-                        RouteNode(r'(\d+)', _SlaveHandler, 'slave')
+                        RouteNode(r'(\d+)', _SlaveHandler, 'slave'),
```

Comment:
You probably have already considered this, but why don't we have a shutdown endpoint nested within the slave id (e.g. v1/slave/23/shutdown) with no body to shut down an individual slave?